frontend/src/components/summary/index.tsx

```diff
@@ -1,29 +1,86 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import differenceInMinutes from 'date-fns/differenceInMinutes';
 import DatePicker from './DatePicker';
+import PieChart from './PieChart';
 
 import summaryStyle from './summary.module.scss';
+import { selectStaff, fetchStaff } from '../../features/staffSlice';
+import {
+  selectEngagement,
+  fetchEngagements,
+} from '../../features/engagementSlice';
+import { useAuth0 } from '../../react-auth0-spa';
+import { StaffData } from '../../constants/employees';
+import { EngagementData } from '../../constants/engagements';
+import capitalizeFirstLetter from '../../utils/string';
+
+const getStaffJob = (staffId: number) => (staffList: StaffData[]) => {
+  const staffMember = staffList.find((staff) => staff.id === staffId);
+  return staffMember ? staffMember.jobTitle : undefined;
```

Comment:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

Would mean result could be null or undefined but I guess you'd just check truthy anyway

Reply:
Nice, I didn't realise this was a thing!

Ideally I wanted to using something like an `Optional` in Java (sometimes called `Maybe`) and the closest thing to this I could find was https://folktale.origamitower.com/api/v2.3.0/en/folktale.maybe.html
but this package doesn't seem to go well with typescript :disappointed: